DABI.js

```diff
@@ -0,0 +1,677 @@
+                var b;
+                while ((b = e.iterateNext())) {
+                    c.push(b)
+                }
+            }
+        }
+        c = this._applyFilters(c, f);
+        if (c.length == 0) {
+            return false
+        } else {
+            return c
+        }
+    }
+};
+FW._Xpath.prototype = new FW._StringMagic();
+FW.detectWeb = function(e, b) {
+    for (var c in FW._scrapers) {
+        var d = FW._scrapers[c];
+        var f = d.evaluateThing(d.itemType, e, b);
+        var a = d.evaluateThing(d.detect, e, b);
+        if (a.length > 0 && a[0]) {
+            return f
+        }
+    }
+    return undefined
+};
+FW.getScraper = function(b, a) {
+    var c = FW.detectWeb(b, a);
+    return FW._scrapers.filter(function(d) {
+        return (d.evaluateThing(d.itemType, b, a) == c) && (d.evaluateThing(d.detect, b, a))
+    })[0]
+};
+FW.doWeb = function(c, a) {
+    var b = FW.getScraper(c, a);
+    b.makeItems(c, a, [], function(f, e, g, d) {
+        e.callHook("scraperDone", f, g, d);
+        if (!f.title) {
+            f.title = ""
+        }
+        f.complete()
+    }, function() {
+        Zotero.done()
+    });
+    Zotero.wait()
+};
+
+function detectWeb(doc, url) {
+    //namespace code
+    var namespace = doc.documentElement.namespaceURI;
+    var nsResolver = namespace ? function(prefix) {
+        if (prefix == 'x') return namespace;
+        else return null;
+    } : null;
+
+    if (doc.title.match("DABI. Datensatz Vollanzeige")) {
+        return "journalArticle";
+    } else if (doc.title.match("DABI: Rechercheergebnis")) {
+        var treffer = doc.evaluate('//*[text()="Keine Treffer für die Suche nach:"]', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext();
+        if (treffer == null) {
+            return "multiple";
+        }
+    }
+}
+
+function doWeb(doc, url) {
+    //namespace code
+    var namespace = doc.documentElement.namespaceURI;
+    var nsResolver = namespace ? function(prefix) {
+        if (prefix == 'x') return namespace;
+        else return null;
+    } : null;
+
+    //variable declarations
+    var articles = new Array();
+    var items = new Object();
+    var nextLink;
+
+    //If Statement checks if page is a Search Result, then saves requested Items
+    if (detectWeb(doc, url) == "multiple") {
+        var links = doc.evaluate('//tr/td[1]/a', doc, nsResolver, XPathResult.ANY_TYPE, null);
+        var title = doc.evaluate('//tr/td[3]', doc, nsResolver, XPathResult.ANY_TYPE, null);
+
+        while (nextLink = links.iterateNext()) {
+            items[nextLink.href] = title.iterateNext().textContent;
+
+        }
+        items = Zotero.selectItems(items);
+        for (var i in items) {
+            articles.push(i);
+        }
+
+    } else if (detectWeb(doc, url) == "journalArticle") {
+        //saves single page items
+        articles = [url];
+    } else {
+        return false;
```

Comment:
This should never be the case, because `detectWeb` should always guarantee that there is something to scrape.